test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/ExecutionPathSegment.java

```diff
@@ -198,7 +210,10 @@ public void insertExecutionStep(final Random random, final AbstractExecutionStep
     }
 
     // add at index will move all elements to the right, including the object on the given index
-    scheduledSteps.add(index, new ScheduledExecutionStep(newStep, newStep, successor.getStep()));
+    scheduledSteps.add(
+        index,
+        new ScheduledExecutionStep(
+            successor.getLogicalPredecessor(), newStep, successor.getStep()));
```

Comment:
Took me a while to understand that, but I have no idea how to improve that situation.

Reply:
I think the issue I have is that I'm not 100% sure what the difference is between execution and logical step and whether we can use here for both the same? If this is the case then we don't need to remove the successor in line 283, and leave the step at that index as it is?

Reply:
discussed face 2 face

Concept is described here: https://github.com/camunda-cloud/zeebe/issues/6568 

Link to concept is available in `ScheduledExecutionStep`